ros2_graph/__main__.py

```diff
@@ -65,60 +149,90 @@ def main():
-            "end",
-            "end",
-        ]
+    mermaid_graph, links_ranges = graph_generator.get_mermaid()
+
+    links_str = style_config["links_str"]
+    shapes = style_config["shapes"]
+    mermaid_convention = (
+        "\n".join(
+            [
+                "subgraph keys[<b>Keys<b/>]",
+                "subgraph nodes[<b><b/>]",
+                "topicb((No connected)):::bugged",
+                f"main_node{shapes[ElementType.MAIN][0]}main{shapes[ElementType.MAIN][1]}:::main",
+                "end",
+                "subgraph connection[<b><b/>]",
+                f"node1{shapes[ElementType.NODE][0]}node1{shapes[ElementType.NODE][1]}:::node",
+                f"node2{shapes[ElementType.NODE][0]}node2{shapes[ElementType.NODE][1]}:::node",
+                f"node1 {links_str[LinkType.SERVICE_SERVER]}|to server| service{shapes[ElementType.SERVICE][0]}Service<br>service/Type{shapes[ElementType.SERVICE][1]}:::service",
+                f"service {links_str[LinkType.SERVICE_CLIENT]}|to client| node2",
+                f"node1 {links_str[LinkType.TOPIC_PUBLISHER]}|publish| topic{shapes[ElementType.TOPIC][0]}Topic<br>topic/Type{shapes[ElementType.TOPIC][1]}:::topic",
+                f"topic {links_str[LinkType.TOPIC_SUBSCRIBER]}|subscribe| node2",
+                f"node1 {links_str[LinkType.ACTION_SERVER]}|to server| action{shapes[ElementType.ACTION][0]}/Action<br>action/Type/{shapes[ElementType.ACTION][1]}:::action",
+                f"action {links_str[LinkType.ACTION_CLIENT]}|to client| node2",
+                "end",
+                "end",
+            ]
+        )
+        if style_config["display_keys"]
+        else ""
     )
 
-    # Add action links of conventions sub graph (the 4th and the 5th)
-    action_links.extend([links_count + 4, links_count + 5])
-    action_links_style = (
-        "linkStyle " + ",".join(map(str, action_links)) + " fill:none,stroke:green;"
+    links_numbers = {
+        link_type: list(range(*ranges))
+        for link_type, ranges in links_ranges.items()
+        if ranges[1] - ranges[0] > 0
+    }
+
+    # Add keys box links
+    if style_config["display_keys"]:
+        last_link = links_ranges[LinkType.ACTION_CLIENT][1]
+        if LinkType.TOPIC_PUBLISHER in links_numbers:
+            links_numbers[LinkType.TOPIC_PUBLISHER].append(last_link + 2)
+        if LinkType.TOPIC_SUBSCRIBER in links_numbers:
+            links_numbers[LinkType.TOPIC_SUBSCRIBER].append(last_link + 3)
+        if LinkType.SERVICE_SERVER in links_numbers:
+            links_numbers[LinkType.SERVICE_SERVER].append(last_link)
+        if LinkType.SERVICE_CLIENT in links_numbers:
+            links_numbers[LinkType.SERVICE_CLIENT].append(last_link + 1)
+        if LinkType.ACTION_SERVER in links_numbers:
+            links_numbers[LinkType.ACTION_SERVER].append(last_link + 4)
+        if LinkType.ACTION_CLIENT in links_numbers:
+            links_numbers[LinkType.ACTION_CLIENT].append(last_link + 5)
```

Comment:
Can you add a comment on what's going on here? Why those numbers

Reply:
Is now clear?

Reply:
thankxs!